### PR TITLE
Fix EventedFileUpdateChecker through a symlink 

### PR DESCRIPTION
**Pull request by jhawthorn**
Locally, on MacOS when I ran `evented_file_update_checker_test.rb`, the "updated should become true when nonexistent directory is added later" test would fail. This is failing because on MacOS `/var` is a symlink to `/private/var`, and this specific test was using `Dir.mktmpdir` explicitly - other tests were making a tmpdir inside `test/` (which maybe we should also change 🤔).

When watching through a symlink, Listen seems to return the absolute path to the modified file. Because `@dirs` references the path of the symlink, its `realpath`, `watching?(f)` returns false.

This PR fixes the issue by replacing each `@dirs` key with its `realpath` if it exists. This done on `boot!`, so it will work with newly created directories through a symlink.

### CODE REVIEW

**Comment by kaspth** [comment]
> other tests were making a tmpdir inside `test/` (which maybe we should also change 🤔).

Seems like a worthy investigation 👍

File: activesupport/lib/active_support/evented_file_update_checker.rb [hunk 1]
```diff
@@ -107,13 +107,20 @@ def execute_if_updated
...
       end
 
       def shutdown!
         Listen.stop
       end
 
+      def normalize_dirs!
+        @dirs.transform_keys! do |dir|
+          dir.exist? ? dir.realpath : dir
+        end
```
Comment (kaspth) [inline activesupport/lib/active_support/evented_file_update_checker.rb]: Does the thing about too many sys calls in Action View apply here? Should we try to spare some SysCalls with something like this?

```ruby
@dirs.transform_keys! do |dir|
  dir.realpath
rescue
  dir
end
```

**Review by kaspth — APPROVED** [review]

File: activesupport/lib/active_support/evented_file_update_checker.rb (hunk 1 above)
Comment (jhawthorn) [reply activesupport/lib/active_support/evented_file_update_checker.rb]: Good question. I don't think that's necessary. `@dirs` are only the top-level dirs being watched, and we probably only run this once on boot (unless a dir is missing and then is created, which should be very rare).

```
> ObjectSpace.each_object(ActiveSupport::EventedFileUpdateChecker).sum{ |x| x.instance_variable_get(:@dirs).size }
# => 28
```

Comment (kaspth) [reply activesupport/lib/active_support/evented_file_update_checker.rb]: Right, so that would be `28 * 2 = 56 syscalls`. Probably not worth it. 👍

**Pull request merged**